src/composables/accounts.ts

```diff
@@ -222,12 +249,83 @@ export function useAccounts() {
     setActiveAccountByProtocolAndIdx(protocol, account?.idx || 0);
   }
 
-  function setMnemonic(newMnemonic: string) {
+  /**
+   * Setting/Resetting the password key logs the user in/out.
+   */
+  function setPasswordKey(key: IKey | null) {
+    passwordKey.value = key;
+    if (IS_EXTENSION) {
+      if (key) {
+        startSession(key, secureLoginTimeout.value);
+      } else {
+        endSession();
+      }
+    }
+  }
+
+  async function openLoginModal() {
+    setLoaderVisible(true);
+    const sessionKey = await getSessionKey();
+    if (sessionKey) {
+      setPasswordKey(sessionKey);
+      const { getAeSdk } = useAeSdk();
+      await getAeSdk();
+      setLoaderVisible(false);
+      return;
+    }
+    setLoaderVisible(false);
+
+    const { openModal } = useModals();
+
+    await openModal(MODAL_PASSWORD_LOGIN);
+    if (!passwordKey.value) {
+      throw new Error('passwordKey was not set after login.');
+    }
+  }
+
+  async function setPassword(newMnemonic: string, password: string) {
```

Comment:
Setters should only set things and avoid doing side things. Please rename the function or make it more functional-programming-friendly :)